singer/src/main/java/com/pinterest/singer/writer/s3/S3Writer.java

```diff
@@ -0,0 +1,438 @@
+package com.pinterest.singer.writer.s3;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.pinterest.singer.common.LogStream;
+import com.pinterest.singer.common.LogStreamWriter;
+import com.pinterest.singer.common.SingerMetrics;
+import com.pinterest.singer.common.errors.LogStreamWriterException;
+import com.pinterest.singer.metrics.OpenTsdbMetricConverter;
+import com.pinterest.singer.thrift.LogMessage;
+import com.pinterest.singer.thrift.LogMessageAndPosition;
+import com.pinterest.singer.thrift.configuration.S3WriterConfig;
+import com.pinterest.singer.utils.SingerUtils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.PosixFilePermission;
+import java.nio.file.attribute.PosixFilePermissions;
+
+import java.util.Set;
+import java.util.List;
+import java.util.Date;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.text.SimpleDateFormat;
+
+import software.amazon.awssdk.services.s3.S3Client;
+
+import static com.pinterest.singer.utils.SingerUtils.getHostnamePrefixes;
+
+/**
+ * A LogStreamWriter for Singer that writes to S3 (writer.type=s3).
+ * */
+public class S3Writer implements LogStreamWriter {
+    public static final String HOSTNAME = SingerUtils.HOSTNAME;
+    private static final Logger LOG = LoggerFactory.getLogger(S3Writer.class);
+    private static final SimpleDateFormat formatter = new SimpleDateFormat("yyyyMMddHHmmssSSS");
+    protected final LogStream logStream;
+    private final String BUFFER_DIR;
+    private final int mB = 1024 * 1024;
```

Comment:
1. Use a self-explanatory name.
2. Since this is a constant, let's follow the variable naming format below.

Maybe `BYTES_IN_MB`?

Reply:
ack, I have made all static final variables with the correct naming format